### DISCUSSION TITLE
Is Google calendar linked to kimai calendar?

### DISCUSSION THREAD

**Discussion by seleondevel** [kimai/kimai#4813] [Plugins]
![스크린샷 2024-04-22 181353](https://github.com/kimai/kimai/assets/165639563/5d75bf92-ce00-413a-b17e-823aff5bc356)
![스크린샷 2024-04-22 181247](https://github.com/kimai/kimai/assets/165639563/764168b3-7209-45b2-8d1a-5db1e32e0b4f)

{
  "error": {
    "code": 400,
    "message": "API key not valid. Please pass a valid API key.",
    "errors": [
      {
        "message": "API key not valid. Please pass a valid API key.",
        "domain": "global",
        "reason": "badRequest"
      }
    ],
    "status": "INVALID_ARGUMENT",
    "details": [
      {
        "@type": "type.googleapis.com/google.rpc.ErrorInfo",
        "reason": "API_KEY_INVALID",
        "domain": "googleapis.com",
        "metadata": {
          "service": "calendar-json.googleapis.com"
        }
      }
    ]
  }
}

The API key is activated normally, and the API key and calendar ID are set in /kimai/config/packages/local.yaml,
Kimai calendar is not marked.
The Google Calendar Share setting is enabled in the company domain, but nothing appears in the kimai calendar.

If you set the public use settings, it will be displayed as udefined as below, but it is a calendar that can only be shared by the company, so you cannot set the public use settings like this.

![스크린샷 2024-04-22 181959](https://github.com/kimai/kimai/assets/165639563/acb1e10c-075a-4006-8324-4b982176199e)

I followed the guide below, but nothing shows on the kimai calendar.

https://www.kimai.org/documentation/local-yaml.html

I wonder if kimai google calendar works normally within the company group, please check and reply.

**Comment by seleondevel** [comment]
If I want to insert Google Calendar in Gimai Calendar, do I have to reveal it?
I set the access permission to be enabled (see all the schedule details) on the company domain group account, not open because it's a calendar that I use within the company with a Google workspace account, but I get an error that the api key is not valid in the Kimai calendar, how do I resolve it?

**Reply by seleondevel** [reply]
If I use the calendar publicly above, it shows up as undefined as above, how can I fix this error?

**Reply by seleondevel** [reply]
I can't solve the Google Calendar, can you give me any advice on this?

**Reply by kevinpapst** [reply]
Yes, it is working with this:
```yaml
kimai:
    calendar:
        google:
            api_key: 'jhgfjhgfjhgfjhgfhjgfhjfg'
            sources:
                holidays:
                    id: 'de.german#holiday@group.v.calendar.google.com'
                    color: '#ccc'
```

**Reply by seleondevel** [reply]
![스크린샷 2024-05-17 103733](https://github.com/kimai/kimai/assets/165639563/b9adc2c3-e94e-4ff4-9c31-0c127af30621)

**Reply by seleondevel** [reply]
![스크린샷 2024-05-17 103544](https://github.com/kimai/kimai/assets/165639563/b9075580-e157-425b-8d8a-931303124112)

**Reply by seleondevel** [reply]
Even though I set up the Api Key properly in the Kimai calendar, I can't see anything on the Google Gallinder schedule.
